haystack/agents/base.py

```diff
@@ -85,6 +86,8 @@ def run(self, tool_input: str, params: Optional[dict] = None) -> str:
             result = self.pipeline_or_node.run(query=tool_input, params=params)
         elif isinstance(self.pipeline_or_node, BaseRetriever):
             result = self.pipeline_or_node.run(query=tool_input, root_node="Query")
+        elif callable(self.pipeline_or_node):
+            result = self.pipeline_or_node(tool_input)
```

Comment:
These lines are not tested

Reply:
I thought I tested it in `test_callable_tool` unit test of `test/agents/test_tools_manager.py` Let me know if I'm mistaken.

Reply:
All good, it's line 92 that's not covered https://coveralls.io/builds/60197848/source?filename=agents%2Fbase.py#L92 we can live with that